azurelinuxagent/ga/policy/policy_engine.py

```diff
@@ -15,24 +15,48 @@
 # Requires Python 2.4+ and Openssl 1.0+
 #
 
+import json
+import os
 from azurelinuxagent.common import logger
 from azurelinuxagent.common.event import WALAEventOperation, add_event
 from azurelinuxagent.common import conf
 from azurelinuxagent.common.exception import AgentError
+from azurelinuxagent.common.protocol.extensions_goal_state_from_vm_settings import _CaseFoldedDict
+from azurelinuxagent.common.osutil.default import DefaultOSUtil
+
+
+# Customer-defined policy is expected to be located at this path.
+# If there is no file at this path, default policy will be used.
+CUSTOM_POLICY_PATH = DefaultOSUtil().get_custom_policy_file_path()
+
+# Default policy values to be used when no custom policy is present.
+_DEFAULT_ALLOW_LISTED_EXTENSIONS_ONLY = False
+_DEFAULT_SIGNATURE_REQUIRED = False
 
 
 class PolicyError(AgentError):
     """
     Error raised during agent policy enforcement.
     """
+    # TODO: split into two error classes for internal/dev errors and user errors.
+    def __init__(self, msg=None, inner=None, code=-1):
```

Comment:
how are we planning to use 'code'?

should probably not have a default value

Reply:
I set the code here to eventually use in create_status_file_if_not_exist(), when we block the extension. But I've removed it for now, I can add in the next PR if needed.